operators/pkg/controller/elasticsearch/driver/downscale.go

```diff
@@ -249,17 +249,42 @@ func updateZenSettingsForDownscale(ctx downscaleContext, downscale ssetDownscale
 		return nil
 	}
 
-	// TODO: only update in case 2->1 masters.
-	// Update Zen1 minimum master nodes API, accounting for the updated downscaled replicas.
-	_, err := zen1.UpdateMinimumMasterNodes(ctx.k8sClient, ctx.es, ctx.esClient, actualStatefulSets, ctx.reconcileState)
-	if err != nil {
+	// Maybe update zen1 minimum_master_nodes.
+	if err := maybeUpdateZen1ForDownscale(ctx, actualStatefulSets); err != nil {
 		return err
 	}
 
-	// Update zen2 settings to exclude leaving master nodes from voting.
+	// Maybe update zen2 settings to exclude leaving master nodes from voting.
 	if err := zen2.AddToVotingConfigExclusions(ctx.esClient, downscale.statefulSet, downscale.leavingNodeNames()); err != nil {
 		return err
 	}
 
 	return nil
 }
+
+// maybeUpdateZen1ForDownscale updates zen1 minimum master nodes if we are downscaling from 2 to 1 master node.
+func maybeUpdateZen1ForDownscale(ctx downscaleContext, actualStatefulSets sset.StatefulSetList) error {
+	if !zen1.AtLeastOneNodeCompatibleWithZen1(actualStatefulSets) {
+		return nil
+	}
+	// If we are moving from 2 to 1 master nodes, we need to update minimum_master_nodes before removing
+	// the 2nd node, otherwise the cluster won't be able to form anymore.
+	// This is inherently unsafe (can cause split brains), but there's no alternative.
+	// For other situations (eg. 3 -> 2), it's fine to update minimum_master_nodes after the node is removed
+	// (will be done at next reconciliation, before nodes removal).
+
+	// retrieve the number of masters currently running ready
+	actualPods, err := sset.GetActualPodsForCluster(ctx.k8sClient, ctx.es)
+	if err != nil {
+		return err
+	}
+	mastersReady := reconcile.AvailableElasticsearchNodes(label.FilterMasterNodePods(actualPods))
```

Comment:
I would not rely on the state of a Pod as it is surfaced by the kubelet.
For example:
* Initial state: 3 masters, `m_m_n` is 2
* Something bad happens on a K8S node, for example the `kubelet` crashes
* Pod is reported as "Unknown" after a few minutes but it still part of the Elasticsearch cluster.
* If for some reason a scale down to 2 is requested then `m_m_n` is set to 1 while there are 2 masters which can lead to a split brain situation.

Reply:
👍 That makes sense, thanks for double-checking.
Some situations could make the cluster unavailable but I think we prefer the cluster being unavailable than the risk of causing split brains.